src/AppBundle/Services/CategoryManager.php

```diff
@@ -29,12 +29,26 @@ public function createCategory($params)
         $category = new Category();
```

Comment:
- exception handling -> Les manager doivent throw'er des expcetions que les controlleurs catch. 

Un exemple d'exception : Unique constraints, database down, price is not a valid number, ....

Reply:
@ahaizoune Unique constraint & price is not a valid number is managed by the validator, in my opinion there is no need to catch an exception for it. What do you think ?

Reply:
Thanks @ahaizoune 
As for input they're usually handled by the validator as @Reddaa said, other than that we write in the controller
```php
try{
    $someManager->someAction($params);
} catch(Exception $e) {
   return $e;
}
```
Is that right? if so which cases should be thrown on the Manager side, i suppose DB failure, missing service or wrong $params passed.

Reply:
@Reddaa une fois le validator catch une erreur qu'est ce qui se passe ?

Reply:
@ahaizoune @coffeemakesmelovelife Une fois le validator catch une erreur let's say sur un formulaire, ca retourne le message de validation pour le field qui n'a pas été validé.
Pour EntityNotFoundException et DatabaseDownException je suis d'accord.

Reply:
@Reddaa dacc.
L'idée est de baser le code sur des exceptions ce qui est plus lisible.
Exemple : 
Au lieu de faire if (x == null) => return null 
On peut imaginer if (x == null) throw NullException ... apres tu catch l'exception et tu la geres comme tu veux